4.0.0-beta1 ninja build fails on nixos due to a potential null dereference

I'm preparing the nixos package for the 4.0.0 update, and I get this error when building from source:

```
[...]
[322/345] Building CXX object src/CMakeFiles/warzone2100.dir/structure.cpp.op.o.cpp.o
FAILED: src/CMakeFiles/warzone2100.dir/structure.cpp.o
/nix/store/i9a3zzf2spmw67jppvkj2yavnbka62id-gcc-wrapper-10.2.0/bin/g++ -DCURL_OPENSSL_DOES_NOT_REQUIRE_LOCKS_INIT -DENABLE_DISCORD -DHAVE_CONFIG_H -DJSON_USE_IMPLICIT_CONVERSIONS=0 -DYY_NO_UNISTD_H -I. -I../. -I../3rdparty -I../3rdparty/date/include -I../3rdparty/discord-rpc/include -I../3rdparty/glad/include -I../3rdparty/launchinfo/include -I../3rdparty/EmbeddedJSONSignature/include -I../3rdparty/quickjs -I../3rdparty/re2 -I../3rdparty/SQLiteCpp/include -isystem ../3rdparty/glm -fstack-protector-strong -fstack-clash-protection -Wpedantic -Wall -Wextra -fno-common -fno-math-errno -Wcast-align -Wcast-qual -Wctor-dtor-privacy -Winit-self -Woverloaded-virtual -Wstrict-null-sentinel -Wwrite-strings -Wpointer-arith -Wmissing-noreturn -Wnon-virtual-dtor -Wnull-dereference -Wduplicated-cond -Walloc-zero -Walloca-larger-than=1024 -Wrestrict -Wfloat-conversion -Wformat-security -Wstringop-truncation -Werror -Wno-logical-op -Wno-float-conversion -Wno-sign-compare -Wno-unused-parameter -Wno-format-truncation -O3 -DNDEBUG -fPIE -pthread -std=c++11 -MD -MT src/CMakeFiles/warzone2100.dir/structure.cpp.o -MF src/CMakeFiles/warzone2100.dir/structure.cpp.o.d -o src/CMakeFiles/warzone2100.dir/structure.cpp.o -c ../src/structure.cpp
../src/structure.cpp: In function 'UDWORD structPowerToBuildOrAddNextModule(const STRUCTURE*)':
../src/structure.cpp:6746:19: error: potential null pointer dereference [-Werror=null-dereference]
 6746 |   return psStats->powerToBuild; // return the cost to build the module
      |                   ^~~~~~~~~~~~
cc1plus: all warnings being treated as errors
ninja: build stopped: subcommand failed.
```

**Your System:**
 - OS: NixOS 20.09
 - Game version: 4.0.0-beta1

--- a/src/structure.cpp
+++ b/src/structure.cpp
@@ -6743,7 +6743,11 @@ UDWORD structPowerToBuildOrAddNextModule(const STRUCTURE *psStruct)
 	if (psStruct->capacity)
 	{
 		STRUCTURE_STATS *psStats = getModuleStat(psStruct);
-		return psStats->powerToBuild; // return the cost to build the module
+		ASSERT(psStats != nullptr, "getModuleStat returned null");
+		if (psStats)
+		{
+			return psStats->powerToBuild; // return the cost to build the module
+		}
 	}
 	// no module attached so building the base structure
 	return psStruct->pStructureType->powerToBuild;

